mycroft/tts/__init__.py

```diff
@@ -209,7 +209,7 @@ def end_audio(self):
 
         self.bus.emit(Message("recognizer_loop:audio_output_end"))
         # Clean the cache as needed
-        cache_dir = mycroft.util.get_cache_directory("tts")
+        cache_dir = mycroft.util.get_cache_directory("tts/" + self.voice)
```

Comment:
`self.voice` can be `None` which would be an issue here can `type(self).__name__` to get the name of the class.

The cache is cleared when a new TTS is created I believe so changing voice should be handled